File: m_storage_v2/mMVVM/loopwiki_java/src/main/java/com/example/gottgried/loopwiki_java/database/models/Note.java
package com.example.gottgried.loopwiki_java.database.models;

import android.arch.persistence.room.PrimaryKey;
import android.arch.persistence.room.TypeConverters;

import com.example.gottgried.loopwiki_java.utils.DateConverter;

import java.util.Date;

// Entity class model of room database
public class Note {
    // room database entity primary key
    @PrimaryKey(autoGenerate = true)
    public int id;
    private String noteTitle;
    private String noteDescription;
    //type converter for date
    @TypeConverters(DateConverter.class)
    private Date createdAt;

    public Note(String noteTitle, String noteDescription, Date createdAt) {
        this.noteTitle = noteTitle;
        this.noteDescription = noteDescription;
        this.createdAt = createdAt;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNoteTitle() {
        return noteTitle;
    }

    public void setNoteTitle(String noteTitle) {
        this.noteTitle = noteTitle;
    }

    public String getNoteDescription() {
        return noteDescription;
    }

    public void setNoteDescription(String noteDescription) {
        this.noteDescription = noteDescription;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }
}

File: m_storage_v2/mMVVM/loopwiki/src/main/java/com/example/gottgried/loopwiki/database/models/Note.kt
package com.example.gottgried.loopwiki.database.models

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import android.arch.persistence.room.TypeConverters
import com.example.gottgried.loopwiki.utils.DateConverter
import java.util.*

// Entity class model of room database
@Entity
data class Note(
        // room database entity primary key
        @PrimaryKey(autoGenerate = true)
        val id: Int,
        val noteTitle: String,
        val noteDescription: String,

        @TypeConverters(DateConverter::class)
        val createdAt: Date
)


//private String noteTitle;
//private String noteDescription;
////type converter for date
//@TypeConverters(DateConverter.class)
//        private Date createdAt;